### DISCUSSION TITLE
Installation error.

### DISCUSSION THREAD

**Discussion by avinav135** [projectdiscovery/chaos-client#302] [Q&A]
Even after installing the chaos project client, when I am trying to run the software using "chaos -h" I am getting a prompt for some other software. Even installed this software but this is not at all related to project discovery so deleted the package.
![Screenshot from 2024-07-20 02-11-44](https://github.com/user-attachments/assets/6fb154be-b180-4676-9fe8-c3ac8f226c09)

**Comment by ehsandeep** [comment]
@avinav369 you can use https://github.com/projectdiscovery/pdtm to install / update tools from projectdiscovery

**Comment by serialfuzzer** [comment]
Most likely this is an issue with environment variable. When I'm lazy I do this:
1. Run the command `find / -name chaos 2>/dev/null`
2. It will return a path like /go/bin/chaos, I copy `/go/bin`
3. edit the file located at /home/myusername/.bashrc
4. add `export PATH=$PATH:/go/bin` in the last new line
remember I got /go/bin in step 2
5. save the file and exit
6. open a new terminal and run chaos (it works like magic)
sorry if it's hard to follow

**Comment by bad-antics** [comment]
You have a naming conflict - there's another package called `chaos` on some systems (a mathematical chaos visualization tool).

**Solution - Install to a different path or rename:**

1. **Use full path:**
```bash
go install github.com/projectdiscovery/chaos-client/cmd/chaos@latest
# Then run with full path:
~/go/bin/chaos -h
```

2. **Create an alias:**
```bash
alias pd-chaos="$HOME/go/bin/chaos"
# Add to ~/.bashrc or ~/.zshrc
```

3. **Check which chaos is running:**
```bash
which chaos
# Shows: /usr/bin/chaos (system chaos)

type chaos
# Might show an alias
```

4. **Priority fix - Add Go bin first in PATH:**
```bash
# Add to ~/.bashrc or ~/.zshrc
export PATH="$HOME/go/bin:$PATH"
```
Then `source ~/.bashrc` and try again.

5. **Remove conflicting package:**
```bash
# If you installed via apt/dnf
sudo apt remove chaos
# or
sudo dnf remove chaos
```

**Verify correct binary:**
```bash
~/go/bin/chaos -version
# Should show: chaos version X.X.X
```

The ProjectDiscovery chaos client will work once the Go bin directory takes precedence in your PATH.

**Comment by bad-antics** [comment]
You have a naming conflict - there's another package called `chaos` on some systems (a mathematical chaos visualization tool).

**Solution - Install to a different path or rename:**

1. **Use full path:**
```bash
go install github.com/projectdiscovery/chaos-client/cmd/chaos@latest
# Then run with full path:
~/go/bin/chaos -h
```

2. **Create an alias:**
```bash
alias pd-chaos="$HOME/go/bin/chaos"
# Add to ~/.bashrc or ~/.zshrc
```

3. **Check which chaos is running:**
```bash
which chaos
# Shows: /usr/bin/chaos (system chaos)

type chaos
# Might show an alias
```

4. **Priority fix - Add Go bin first in PATH:**
```bash
# Add to ~/.bashrc or ~/.zshrc
export PATH="$HOME/go/bin:$PATH"
```
Then `source ~/.bashrc` and try again.

5. **Remove conflicting package:**
```bash
# If you installed via apt/dnf
sudo apt remove chaos
# or
sudo dnf remove chaos
```

**Verify correct binary:**
```bash
~/go/bin/chaos -version
# Should show: chaos version X.X.X
```

The ProjectDiscovery chaos client will work once the Go bin directory takes precedence in your PATH.

**Comment by bad-antics** [comment]
The issue is that another package named `chaos` exists in some Linux package managers, and it's shadowing the ProjectDiscovery chaos-client binary.

**Fix — install the correct binary:**

1. **Using Go (recommended):**
```bash
go install -v github.com/projectdiscovery/chaos-client/cmd/chaos@latest
```
Make sure `$GOPATH/bin` (usually `~/go/bin`) is in your PATH:
```bash
export PATH=$PATH:$(go env GOPATH)/bin
```

2. **Using pdtm (ProjectDiscovery Tool Manager):**
```bash
go install -v github.com/projectdiscovery/pdtm/cmd/pdtm@latest
pdtm -install chaos-client
```

3. **Direct download:**
```bash
# Linux amd64
curl -sL https://github.com/projectdiscovery/chaos-client/releases/latest/download/chaos-client_linux_amd64.zip -o chaos.zip
unzip chaos.zip
sudo mv chaos /usr/local/bin/chaos
```

4. **Check which chaos is being called:**
```bash
which chaos
# If it shows /usr/bin/chaos or similar system path, the wrong one is first in PATH

# Fix by putting Go bin first
export PATH=$(go env GOPATH)/bin:$PATH
```

5. **If the system `chaos` package was installed via apt:**
```bash
sudo apt remove chaos  # Remove the conflicting package
```

Then verify:
```bash
chaos -version
# Should show: chaos-client version x.x.x
```

The key issue is PATH priority — make sure `~/go/bin` comes before `/usr/bin` in your PATH.
